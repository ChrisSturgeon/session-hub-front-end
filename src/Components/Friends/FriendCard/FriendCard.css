.friend-card {
  width: clamp(200px, 100%, 450px);
  display: flex;
  gap: 1em;
  align-items: center;
  padding: 0.5em 1em;
  background-color: var(--bg-grey);
  border-radius: 10px;
  text-decoration: none;
  color: var(--dark-blue);
  box-shadow: 0px 3px 15px rgba(0, 0, 0, 0.2);
  transition: transform 150ms ease-in-out;
  font-size: 1.1rem;
  font-weight: 500;
}

.friend-card img {
  max-height: 70px;
  border-radius: 100vw;
}

.friend-card:hover {
  position: relative;
  transform: translateX(10px);
}

.profile-dummy-img {
  height: 70px;
  width: 70px;

  border-radius: 100%;
  background-color: lightblue;
}
