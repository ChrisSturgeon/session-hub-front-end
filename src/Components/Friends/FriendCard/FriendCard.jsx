import './FriendCard.css';
import { Link } from 'react-router-dom';

export default function FriendCard({ friendData }) {
  const profileURL = `/profile/${friendData.name}`;
  return (
    <Link
      to={profileURL}
      state={{ profileData: friendData }}
      className="friend-card"
    >
      <div className="profile-img"></div>
      <div>{friendData.name}</div>
    </Link>
  );
}
