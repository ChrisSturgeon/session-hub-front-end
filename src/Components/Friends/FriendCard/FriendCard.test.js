import React from 'react';
import { getByTestId, render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { BrowserRouter } from 'react-router-dom';
import FriendCard from './FriendCard';

const mockFriendData = {
  username: 'Chris Sturgeon',
  _id: '1ab2c3',
  thumbURL: 'https://i.imgur.com/RpjwdOf.png',
};

describe('Friend Card', () => {
  const setup = () => {
    render(
      <BrowserRouter>
        <FriendCard friendData={mockFriendData} />
      </BrowserRouter>
    );
  };

  it('renders', () => {
    setup();
    expect(screen.getByRole('link')).toBeInTheDocument();
  });

  it('Matches snapshot', () => {
    const { container } = render(
      <BrowserRouter>
        <FriendCard friendData={mockFriendData} />
      </BrowserRouter>
    );
    expect(container).toMatchSnapshot();
  });

  it('Links to correct user profile', () => {
    setup();
    expect(screen.getByRole('link')).toHaveAttribute(
      'href',
      '/profile/1ab2c3/posts'
    );
  });

  it('Displays image with correct src from props', () => {
    setup();
    expect(screen.getByRole('img')).toHaveAttribute(
      'src',
      'https://i.imgur.com/RpjwdOf.png'
    );
  });

  it('Displays username from props', () => {
    setup();
    expect(screen.getByTestId('username')).toHaveTextContent('Chris Sturgeon');
  });
});
