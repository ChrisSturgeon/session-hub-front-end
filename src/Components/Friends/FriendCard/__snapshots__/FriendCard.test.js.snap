// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Friend Card Matches snapshot 1`] = `
<div>
  <a
    class="friend-card"
    href="/profile/1ab2c3/posts"
  >
    <img
      alt="display-pic"
      src="https://i.imgur.com/RpjwdOf.png"
    />
    <div
      data-testid="username"
    >
      Chris Sturgeon
    </div>
  </a>
</div>
`;
